knime_extension/src/nodes/spatialtool.py

```diff
@@ -1292,6 +1292,94 @@ def execute(self, exec_context: knext.ExecutionContext, input_table):
         return knut.to_table(grid, exec_context)
 
 
+############################################
+# Point to H3 Node
+############################################
+
+
+@knext.node(
+    name="Point to H3",
+    node_type=knext.NodeType.MANIPULATOR,
+    icon_path=__NODE_ICON_PATH + "PointToH3.png",
+    category=__category,
+    after="",
+)
+@knext.input_table(
+    name="Input Table",
+    description="Input table with the geometry",
+)
+@knext.output_table(
+    name="Output Table",
+    description="Output table with the H3 Cell Indices and hexagons column (optional).",
+)
+class PointToH3:
+    """Point to H3 node.
+    This node returns the H3 Cell Indices and hexagons (optional) for each point in the input table.
+    """
+
+    geo_col = knut.geo_col_parameter()
+
+    zoom = knext.IntParameter(
+        "Zoom level",
+        """The zoom level of the grid from 0 to 15 (default value is 8). The bigger the zoom level, the smaller the 
+        hexagon. If the zoom level is too small, the hexagon might be too big to fit in the input polygon which will
+        result in an error. A very small zoom level might result in a very large output table even for smaller 
+        input polygons. 
+        For more details about the zoom levels  refer to 
+        [Tables of Cell Statistics Across Resolutions.](https://h3geo.org/docs/core-library/restable/)
+        """,
+        default_value=8,
+        min_value=0,
+        max_value=15,
+    )
+
+    _COL_ID = "H3 Cell Index"
+    _COL_GEOMETRY = "geometry"
+
+    def configure(self, configure_context, input_schema):
+        self.geo_col = knut.column_exists_or_preset(
+            configure_context, self.geo_col, input_schema, knut.is_geo_point
+        )
+
+        return knext.Schema.from_columns(
+            [
+                knext.Column(knext.string(), self._COL_ID),
+                knext.Column(input_schema[self.geo_col].ktype, self._COL_GEOMETRY),
+            ]
+        )
+
+    def execute(self, exec_context: knext.ExecutionContext, input_table):
+        import h3
+        import geopandas as gpd
+        import pandas as pd
+        from shapely.geometry import Polygon
+
+        gdf = knut.load_geo_data_frame(input_table, self.geo_col, exec_context)
+
+        knut.check_canceled(exec_context)
+        exec_context.set_progress(0.1, "Projecting input point...")
+        gdf.to_crs(4326, inplace=True)
+
+        knut.check_canceled(exec_context)
+        exec_context.set_progress(0.5, "Computing H3 hexagons...")
+        h3_hexes = gdf.apply(
+            lambda x: h3.geo_to_h3(x.geometry.y, x.geometry.x, self.zoom), axis=1
```

Comment:
Doesn't work if the selected input column is not called geometry

Reply:
It should be work since the geometry column that user selected is already replaced in the above code.

Reply:
OK I will double check and try to reproduce